pkg/fileutils/fileutils.go

```diff
@@ -7,21 +7,50 @@ import (
 )
 
 // Matches returns true if relFilePath matches any of the patterns
+// and isn't excluded by any of the subsequent patterns.
 func Matches(relFilePath string, patterns []string) (bool, error) {
-	for _, exclude := range patterns {
-		matched, err := filepath.Match(exclude, relFilePath)
+
+	matched := false
+
+	for _, pattern := range patterns {
+
+		var negative bool
+		if pattern == "" {
+			continue
+		}
+
+		if pattern[0] == '!' {
+			if len(pattern) == 1 {
+				continue
+			}
+			negative = true
+			pattern = pattern[1:]
+		}
+
+		match, err := filepath.Match(pattern, relFilePath)
 		if err != nil {
-			logrus.Errorf("Error matching: %s (pattern: %s)", relFilePath, exclude)
+			logrus.Errorf("Error matching: %s (pattern: %s)", relFilePath, pattern)
 			return false, err
 		}
-		if matched {
+
+		if match {
+
+			if negative {
+				matched = false
+				continue
+			}
+
 			if filepath.Clean(relFilePath) == "." {
```

Comment:
Should this check for "." apply to the ! case too?
If so, you can move this up before the "if negative" line and then the line below becomes:
```
matched  = !negative
```
and you can remove the "if negative" check.